backend/src/apiserver/client_manager.go

```diff
@@ -66,7 +66,7 @@ type ClientManager struct {
 	dBStatusStore          storage.DBStatusStoreInterface
 	defaultExperimentStore storage.DefaultExperimentStoreInterface
 	objectStore            storage.ObjectStoreInterface
-	wfClient               workflowclient.WorkflowInterface
+	wfClient               map[string]workflowclient.WorkflowInterface
```

Comment:
I think what we should do instead is to create a single client without initializing the namespace, and modify the workflow to inject the namespace at here.
https://github.com/kubeflow/pipelines/blob/4500d244d715ef67e77d94e6cdb62962523792f7/backend/src/apiserver/resource/resource_manager.go#L266

Reply:
This will not work because the argo client is per workflow per namespace. Or am I mistaken?

Reply:
Here is the reference of the argo API
https://github.com/argoproj/argo/blob/master/pkg/client/clientset/versioned/typed/workflow/v1alpha1/workflow_client.go#L22

I think the naming here is a bit misleading. I think what you could do in client_manager is instead of keeping a list of WorkflowInterface, keep a ArgoprojV1alpha1Client. Then you could dynamically use ArgoprojV1alpha1Client to create a namespace specific workflowInterface at run time when handling the api call.

Reply:
Done.